extensions/standard-processors/processors/HashContent.cpp

```diff
@@ -62,11 +63,8 @@ void HashContent::onSchedule(core::ProcessContext *context, core::ProcessSession
   attrKey_ = (context->getProperty(HashAttribute.getName(), value)) ? value : "Checksum";
   algoName_ = (context->getProperty(HashAlgorithm.getName(), value)) ? value : "SHA256";
 
-  if (context->getProperty(HashAlgorithm.getName(), value)) {
-    bool bool_value;
-    failOnEmpty_ = utils::StringUtils::StringToBool(value, bool_value) && bool_value;  // Only true in case of valid true string
-  } else {
-    failOnEmpty_ = false;
+  if (context->getProperty(FailOnEmpty.getName(), value)) {
+    failOnEmpty_ = utils::StringUtils::toBool(value).value_or(false);
```

Comment:
We need to set `failOnEmpty_` to false when `getProperty()` returns false.  You can use either an else branch, or a `getProperty(...) && toBool(...).value_or(false)` expression similar to what you used elsewhere.